### ISSUE TITLE
Pull-To-Reload often results in scrolling left/right

### ISSUE THREAD

**Issue by Drachenkaetzchen** [tuskyapp/Tusky#2112]
When attempting to use pull-To-Reload, tusky often changes to the right page instead. It seems that the angle is very narrow to make pull-To-Reload work,resulting in unwanted page changes, especially when using a phone one-handed. 

I double checked against other apps, and most seem to use another angle. Tusky seems to have its margin at 45° (sometimes even less) where other apps have an angle up to 70° to make pull To refresh work. 

* * * *
- Tusky Version: 14.0
- Android Version: 11
- Android Device: Samsung M31

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by Drachenkaetzchen** [comment]
I tested this a bit further and sometimes it works as expected and sometimes it changes the page even with swiping almost vertically.

I guess this is because the mechanism only respects the very first few pixels to decide whether to scroll/refesh or change page.

**Comment by a1batross** [comment]
It's because of the changes in ViewPager2 library used for tabs. For some reason, they made it very sensitive. 

The only way to fix it is to decrease sensitivity through Java reflection, which is just a nasty hack.

There is an option in settings to disable swipe between tabs, this may help.

**Comment by Drachenkaetzchen** [comment]
A bit of research turned up this stack overflow question:

https://stackoverflow.com/questions/57336292/viewpager2-interaction-with-swiperefreshlayout

One person is suggesting that the solution is to upgrade to v1.1 of ViewPager2. The referenced issue at Google is https://issuetracker.google.com/issues/138314213

However, I do know virtually nothing about Android development, so please feel free to ignore this research :)

**Comment by a1batross** [comment]
Didn't knew there is 1.1 released. Thanks for heads up!

**Comment by Drachenkaetzchen** [comment]
Just in case it doesn't get missed: SwipeRefreshLayout seems to have gotten an update, too, and the nesting seems to be important. Hope that helps :)

**Comment by connyduck** [comment]
You can download a test build with the dependency upgrades included here: https://app.bitrise.io/artifact/67993475/p/ddc6d8fe466319bbc32e4798479f5496

Please test if it indeed fixes the issue.

**Comment by Drachenkaetzchen** [comment]
Unfortunately no, it doesn't fix the issue.

What did improve:
- If swiping down at a slight angle, the page change is indicated but not performed (sensitivity reduced)

What did not improve:
- No reload is indicated

Is there any way to record a sample video which also shows where the touch events took place?

**Comment by connyduck** [comment]
Android Studio can do that

**Comment by Drachenkaetzchen** [comment]
I meant recording it on the device, to actually see what input the Android device actually receives.

**Comment by connyduck** [comment]
Yes, you connect a device with Usb Debugging and then you can record its screen from within Android Studio (in the logcat tab) that also shows the touch events.

**Comment by connyduck** [comment]
Oh, and if you have a screenrecording app on your phone, you can also enable "show visual feedback for taps" in any Androids developer options

**Comment by Modelmat** [comment]
Hi, I also have this issue.

This is a recording on the current release of the app itself:

https://user-images.githubusercontent.com/25718989/117568398-8a3dbf80-b0af-11eb-8b53-ca81f7cdf6eb.mp4

On the test version you posted the issue is about the same.

https://user-images.githubusercontent.com/25718989/117568723-8b6fec00-b0b1-11eb-9e4c-75da54e8b803.mp4

On both apps it seems to be about 10 degrees from the the vertical. The faster you swipe the higher the angle can be.

**Comment by connyduck** [comment]
I opened an issue at Google about this almost a year ago, but it seems nobody cares :(
https://issuetracker.google.com/issues/157335285

Maybe we should do the hacky workaround

**Comment by PitchforkAssistant** [comment]
This still seems to be an issue. I've noticed it even happens when just scrolling the feed. It seems that it happens most when I use my left thumb to lazily scroll the feed, a lot of the swipes end up being curve shaped.


https://user-images.githubusercontent.com/18224088/205486927-46555866-7b43-4f42-9a4f-f820510636a2.mp4


https://user-images.githubusercontent.com/18224088/205487964-0a6a05f6-38ea-4abd-8560-bb911a8e8cb4.mp4

I've just disabled swiping between tabs for now and that works alright.
